cmds/id/id.go

```diff
@@ -0,0 +1,265 @@
+// Copyright 2013-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+// Print process information.
+//
+// Synopsis:
+//      id [-gGnu]
+//
+// Description:
+//      id displays the uid, guid and groups of the calling process
+// Options:
+//  		-g, --group     print only the effective group ID
+//		  -G, --groups    print all group IDs
+//		  -n, --name      print a name instead of a number, for -ugG
+//		  -u, --user      print only the effective user ID
+//
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"syscall"
+)
+
+var (
+	GROUP_FILE  = "/etc/group"
+	PASSWD_FILE = "/etc/passwd"
+	l           = log.New(os.Stderr, "", 0)
+
+	flags struct {
+		g bool
+		G bool
+		n bool
+		u bool
+	}
+)
+
+func correctFlags(flags ...bool) bool {
+	n := 0
+	for _, v := range flags {
+		if v {
+			n += 1
+		}
+	}
+	if n > 1 {
+		return false
+	} else {
+		return true
+	}
+}
+
+func initFlags() error {
+	flag.BoolVar(&flags.g, "g", false, "print only the effective group ID")
+	flag.BoolVar(&flags.G, "G", false, "print all group IDs")
+	flag.BoolVar(&flags.n, "n", false, "print a name instead of a number, for -ugG")
+	flag.BoolVar(&flags.u, "u", false, "print only the effective user ID")
+	flag.Parse()
+	if !correctFlags(flags.G, flags.g, flags.u) {
+		return fmt.Errorf("cannot print \"only\" of more than one choice\n")
+
+	}
+	if flags.n && !(flags.G || flags.g || flags.u) {
+		return fmt.Errorf("cannot print only names in default format\n")
+	}
+
+	return nil
+}
+
+type User struct {
+	Name   string
+	Uid    int
```

Comment:
If I fully capitalize the acronyms then I'll get a runtime error because I have methods and member variables with the same name.